### PR TITLE
DOC:Remove DataFrame.append from the 10min intro

### LINKED ISSUE #27518: Remove DataFrame.append from the 10min intro
We show it https://pandas.pydata.org/pandas-docs/stable/getting_started/10min.html#append.

I don't think we should recommend it there, especially without providing context that the complexity of DataFrame.append is very different from list.append.

### PR DESCRIPTION
**Pull request by sameshl**
Remove the `append` section from 10 min intro doc as complexity of that is very different than `list.append`

- [x] closes #27518

### CODE REVIEW

**Comment by TomAugspurger** [comment]
As a note, the DataFrame.append API docs have a good discussion of this https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.append.html#pandas.DataFrame.append.

We need to determine whether the "append" use-case (incrementally building a DataFrame) is important enough to warrant inclusion in the 10min intro.

**Comment by sameshl** [comment]
I think as we already have explained `concat` at https://pandas.pydata.org/pandas-docs/stable/getting_started/10min.html#concat, it should suffice for a beginner and it is anyway better than iteratively using `append`.

**Comment by WillAyd** [comment]
I'd prefer a warning and link to append docs about performance rather than outright removal

**Comment by TomAugspurger** [comment]
CI should be fixed if you fetch and merge master then push.

> I'd prefer a warning and link to append docs about performance rather than outright removal

Note that we do still discuss DataFrame.append in the user guide: https://pandas.pydata.org/pandas-docs/stable/user_guide/merging.html#concatenating-using-append. I just don't think it's valuable enough to include in your first 10 minutes to learning pandas.

That said, we may want to head of the style of iteratively building a DataFrame, like you would with a list, by leaving a comment at the bottom of https://pandas.pydata.org/pandas-docs/stable/getting_started/10min.html#object-creation saying something like

> Adding a column to a DataFrame is relatively fast. Adding a row, however, requires a full copy of the data and a construction of a new Index, which may be expensive. Rather than iteratively building a DataFrame by appending records, we recommend passing the pre-built list of records to the DataFrame constructor. See ... for more.

How does that sound @WillAyd?

**Comment by WillAyd** [comment]
Sounds good!

File: doc/source/getting_started/10min.rst [hunk 1]
```diff
@@ -83,6 +83,13 @@ As you can see, the columns ``A``, ``B``, ``C``, and ``D`` are automatically
 tab completed. ``E`` is there as well; the rest of the attributes have been
 truncated for brevity.
 
+.. note::
```
Comment (jreback) [inline doc/source/getting_started/10min.rst]: this doesn't seem like a good place for this, can you put near usages of concat instead

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/getting_started/10min.rst (hunk 1 above)
Comment (sameshl) [reply doc/source/getting_started/10min.rst]: Sure, even I felt that would be a better place for this.

**Comment by sameshl** [comment]
@jreback I have made the required changes.

File: doc/source/getting_started/10min.rst [hunk 2]
```diff
@@ -468,6 +468,13 @@ Concatenating pandas objects together with :func:`concat`:
 
    pd.concat(pieces)
 
+.. note::
+   Adding a column to a DataFrame is relatively fast. Adding a row, however,
+   requires a full copy of the data and a construction of a new Index, which
```
Comment (jreback) [inline doc/source/getting_started/10min.rst]: use double-backticks around DataFrame. Also let's make this shorter / simpler.

first part do something like

```
Adding a column to a DataFrame is relatively fast, However, adding a row requires a copy, and may be expensive.
```

Comment (sameshl) [reply doc/source/getting_started/10min.rst]: @jreback Done. Let me know if any more changes are required.

File: doc/source/getting_started/10min.rst [hunk 3]
```diff
@@ -468,6 +468,13 @@ Concatenating pandas objects together with :func:`concat`:
 
    pd.concat(pieces)
 
+.. note::
+   Adding a column to a ``DataFrame`` is relatively fast. However, adding
+   a row requires a copy, and may be expensive. It is recommended to
```
Comment (TomAugspurger) [inline doc/source/getting_started/10min.rst]: "It is recommended to pass" -> "We recommend passing"

File: doc/source/getting_started/10min.rst [hunk 4]
```diff
@@ -468,6 +468,13 @@ Concatenating pandas objects together with :func:`concat`:
...
+   pass a pre-built list of records to the ``Dataframe`` constructor instead
```
Comment (TomAugspurger) [inline doc/source/getting_started/10min.rst]: `Dataframe` to `DataFrame` (capital "F").

**Review by TomAugspurger — APPROVED** [review]
Small style corrections. LGTM otherwise.

**Comment by sameshl** [comment]
@TomAugspurger Done!

**Comment by TomAugspurger** [comment]
Thanks @sameshl!

**Pull request merged**